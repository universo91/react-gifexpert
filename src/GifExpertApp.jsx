import React, { useState } from 'react';
import  AddCategory  from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState( defaultCategories );
    //const [categories, setCategories] = useState(['One punch']);
   
    return (
        <>
            <h2> GifExpertApp </h2>
            <AddCategory setCategories = { setCategories } />
            <hr />        
            
            { 
                categories.map( category => (
                    <GifGrid 
                        key = { category }
                        category = { category }
                    /> 
                ))
            }
            
        </>
    )
}

export default GifExpertApp
