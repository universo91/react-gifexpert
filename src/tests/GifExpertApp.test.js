import { shallow } from "enzyme";
import GifExpertApp from '../GifExpertApp';
describe('Pruebas sobre GifExpertApp', () => {

    test('El componente debe de mostrarse correctamente', () => {
        
        const wrapper = shallow( <GifExpertApp /> );
        expect( wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        
        const categories =['one punch', 'dragon ball'];

        const wrapper = shallow( <GifExpertApp defaultCategories = { categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length).toBe(categories.length);
    })
})