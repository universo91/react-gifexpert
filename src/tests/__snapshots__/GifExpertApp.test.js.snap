// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="one punch"
    key="one punch"
  />
  <GifGrid
    category="dragon ball"
    key="dragon ball"
  />
</Fragment>
`;

exports[`Pruebas sobre GifExpertApp El componente debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
     GifExpertApp 
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
